File: client/src/app/modules/quotation/services/asso.service.field.type.service.ts
import { AppRestService } from "src/app/services/appRest.service";
import { AssoServiceFieldType } from "../model/AssoServiceFieldType";
import { HttpClient, HttpParams } from "@angular/common/http";
import { Injectable } from "@angular/core";

@Injectable({
  providedIn: 'root'
})
export class AssoServiceFieldTypeService extends AppRestService<AssoServiceFieldType> {

  constructor(http: HttpClient) {
    super(http, "quotation");
  }
  addAssoServiceFieldType(serviceId: number, serviceFieldTypeId: number) {
    return this.postItem(new HttpParams().set("serviceId", serviceId).set("serviceFieldTypeId", serviceFieldTypeId), "asso-service-field-type/add", null, "Type de champ ajouté au service", "Erreur lors de l'ajout du type de champ");
  }
}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/AssoServiceFieldTypeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import com.jss.osiris.modules.osiris.quotation.model.AssoServiceFieldType;

public interface AssoServiceFieldTypeService {
    public AssoServiceFieldType getAssoServiceFieldType(Integer id);

    public AssoServiceFieldType addOrUpdateServiceFieldType(AssoServiceFieldType assoServiceFieldType);
}
